### DISCUSSION TITLE
How to solidify Solid Project

### DISCUSSION THREAD

**Discussion by jeff-zucker** [solid/solidproject.org#783] [Announcements]
I began a [discussion on dogfooding solidproject.org](https://github.com/solid/solidproject.org/issues/582) a while ago, but it's time for a fresh look.  My own vision of this looks something like the following model, implemented so that machines can follow it to the RDF and humans  can access it as a website.

![solidproject-ideas](https://user-images.githubusercontent.com/44732708/231565562-1e14894c-f48e-4a28-8c5a-185b6cebda33.png)

Here are some of the issues that need addressing : 

### How do we generate the website?

Currently it is done with Jekyll but as [my recent assessment of creators](https://github.com/solid/process/issues/314) shows, the creators (myself included) are somewhat lacking in knowledge of Jekyll.  I would like to see this eventually be done by our own scripts which generate the site using Solid Technology and taking advantage of existing Solid RDF data such as lists of team members.  I have created several previous versions of such a method, demoed them, got some discussion going. I've applied this to RDF generated HTML-RDFa for team-members, glossaries, repositories, specifications, and ontologies.  My most up to date version of a library  for componentized website/app design - [solid-uix](https://solid-uix)  - uses SolidOS as an example rather than Solid Project, but it can be, I think, adapted.   You might also want to look at an earlier demo of an [RDF-generated solidproject.org](https://jeff-zucker.solidcommunity.net/sp4/).

### How do we edit the website? 

Currently this is done with Github PRs.  Both @timea-solid  and I have worked on parts of this problem.  She has a POC on creating a backup/restore system and I have been developing a series of [editing tools (WiP)](https://jeff-zucker.github.io/solid-content-manager ) which allow editors without RDF knowledge to edit the RDF sources and see in real time how the changes look in their eventual HTML representation using editor selected templates.  This relates to how the Creators edit solidproject.org but also to the more general question of the relationship of front-end developers and content providers with RDF.

### Where is the source of truth and how is it stored?

If we're going to base solidproject.org on RDF we need a plan to know where the RDF is.  Currently it's spread over HTML-RDFa pages, RDF documents on github, RDF on pods and other sources.  Take the Solid Team info - the only actually complete and up-to-date (ish) data is hard-coded in HTML on the website.  There is an incomplete RDF version in one of timbl's address books.  There is the github list of who controls what repo.  You get the picture ... We need some level of standardization and a way that editing the RDF gets reflected on the website rather than the current disconnect between the website and the data it is based on.

### Which Ontologies?

The question of ontologies used on solidproject.org is relevant, see the discussion at https://github.com/solid/team/issues/15 and https://github.com/solid/team/issues/16.  We also need an ontology to represent applications and libraries and pod-providers, etc.

### How should organizations employ RDF to present themselves?

@timbl  has been approaching this question from several directions.  This [article](https://solidos.solidcommunity.net/public/2021/BuildingSolidAppsUsingPublicData-V3.html) touches on how organizations define themselves.  He has introduced a new predicate `solid:community`  which used to interconnect organizations and their members with triples like `<#me> solid:community <someOrganization>` and a plan to use this to, for example expose the contacts lists and meeting-trackers of the organization to its members.  See also [my article](https://solidos.solidcommunity.net/public/2022/modeling-social-benefit-of-organizations/index.html) on the topic of the RDF modeling of organizations.

### The future

Recently @danielbakas has expressed an interest in helping to pursue this vision.  @jaxoncreed has mentioned the possibility of future work on a UI for solidproject.org.  I and the SolidOS team will be pursuing the use of solid-uix as a components/templating system and its use to create access to RDF for front-end developers and content-creators without them needing advanced RDF knowledge.  The Solid Team admins are progressing on moving solidcommunity.net to CSS and solidproject.org to solidcommunity.net.  And ____   [please fill in the blank].

**Comment by sjoerdvangroning** [comment]
What do you think if the website would be hosted in a Solid pod with all the content, html,css, templates and the relevent data would be pulled in from RDF sources? Editing possible via a CMS. Access and editing would be enforced by access to the Solid POD.
Would that be a workable/satisfying model?

**Reply by timea-solid** [reply]
There is this debate about serving HTML from a Pod: https://forum.solidproject.org/t/is-it-secure-for-pods-to-serve-html-files/6379/9

**Comment by csarven** [comment]
https://github.com/solid/solidproject.org/issues/248 + https://github.com/linkeddata/dokieli + magic would be one way.

**Comment by jeff-zucker** [comment]
I beleive the predicates below are sufficient for describing pretty much all of the contents of solidproject.org.  I propose this as the basis of a shape for organizations  and specifically as the shape for Solid Project and Solid Team and as a basis for RDFa on the site itself.   To be addressed later - predicates that Organizations share with Persons, relationships between organizations, and more on types of organization.

### Organization Description
|  | RDF |
| - | - |
| Name |  :SolidTeam **vcard:organization-name** "The Solid Team"
| Purpose | :SolidTeam **org:purpose** :Solid_Vision_Statement.
| Code of Conduct |  :SolidTeam **schema:ethicsPolicy** :Solid_COC.
| Diversity Policy |  :SolidTeam **schema:diversityPolicy** :DEI_Charter.
| Process |  :SolidTeam **schema:publishingPolicy** :Solid_Process.
| Press |  :SolidTeam **schema:subjectOf** [a schema:NewsArticle], [a schema:Video].

### Organization Structure
|  | RDF |
| - | - |
| Departments |  :SolidTeam **org:hasUnit** :Admins.
| Members | :SolidTeam **org:hasMember** :VirginiaB.
| Roles | see the org: ontology.

### Organization Deliverables
|  | RDF |
| - | - |
| Identity Provider | :SolidComDotNet a **pto:Authentication_Server**; schema:provider :SolidTeam.
| Storage Provider | :SolidComDotNet a **pto:Storage_server**;  schema:provider :SolidTeam.
| Forum & Chatrooms | :SolidTeam **sioc_administrator_of** :SolidForum.
| Events | :SolidPractitionersMeetup a **schema:Event**; schema:organizer :SolidTeam.
| Website | :SolidProjectDotOrg a **schema:Website**; schema:maintainer :SolidTeam.
| Specifications | :SolidProtocol a **pto:Specification_(technical_standard)**; schema:maintainer :W3CGG, :SolidTeam.
| Ontologies | :SolidTerms a **owl:ontology**; schema:maintainer :SolidTeam.
| Test Suites | :SolidTestSuite a **pto:Test_suite**; schema:maintainer :SolidTeam.
| Use Cases | :SolidUseCase a **pro:Use_case**; schema:maintainer :SolidTeam.
| Dev Tools | :SolidNamespace a **pto:Programming_tool** schema:maintainer :SolidTeam.
| Projects | :SolidShapesRepo a **doap:Project** schema:maintainer :SolidTeam.

### Orgainzation Stateholders

|  | RDF |
| - | - |
| Community Members | :TimeaT a **schema:Person**.
| Nonprofit Organizations  |  :SleepyBike a **schema:NGO**. 
| University Departments | :Imec a **schema:ResearchOrganization**. 
| Government Agencies | :DigitalFlanders a **schema:GovernmentOrganization**.
| Enterprises | :Janiero a **schema:Corporation**. 

### Stakeholder Deliverables

|  | RDF |
| - | - |
| Identity Server Software | :ESS a **pto:Authentication_Server**; schema:manufacturer :Inrupt.
| Storage Server Software | :Manas  a **pto:Storage_Server**; schema:manufacturer :Damoo.
| Identity Provider | :SolidWebDotMe a **pto:Authentication_Server**; schema:provider :MathiasE.
| Storage Provider | :SolidWebDotMe  a **pto:Storage_Server**; schema:provider :MathiasE.
| Oboarding Resources | :UsingLDO a **pto:Tutorial**; schema:creator :JacksonM.
| Applications | :Dokeli a **schema:SoftwareApplication**; schema:creator :Sarven.
| Dev Tools | :LDO a **pto:Programming_tool**; schema:creator :JacksonM.

### Note

pto: is the prefix for the Product Ontology `http://www.productontology.org/id/. This ontology is a collaboration between Wikipedia, Schema.org, and GoodReations and leverages Wikipedia's many languages.
